### PR TITLE
Upgrade Gradle and improve the project configuration

### PR DESCRIPTION
**Pull request by mridang**
## Description
Upgraded to Gradle 6.7 and added the necessary Github templates

## Motivation and Context
Ensures that we're using the latest version of Gradle and that the project configuration is sound.

## Related Jira ticket
None

## Testing
None.

## Documentation
None.

## Checklist

- [x] Pull Request is properly described.
- [x] The corresponding Jira ticket is updated.
- [x] I have requested a review from at least one reviewer.
- [x] The changes are tested
- [ ] I have verified my UX changes with a designer
- [x] I have checked my code for any possible security vulnerabilities

## Screenshots
None.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+name: Bug report
+about: Report errors and problems
+title: ''
+labels: bug
+assignees: ''
+
+---
+
+**Magento 2 version(s) used**: x.y.z
```
Comment (gary-madden) [inline .github/ISSUE_TEMPLATE/bug_report.md:10]: Copy pasta?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+**Extension version(s) affected**: x.y.z
```
Comment (gary-madden) [inline .github/ISSUE_TEMPLATE/bug_report.md:11]: Copy pasta?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+**How to reproduce**  
+<!-- Code, config and/or steps needed to reproduce the problem. -->
+
+**Magento 2 mode**
+- [ ] Developer 
+- [ ] Production
+
+**Full page cache**
+- [ ] Enabled 
+- [ ] Disabled
```
Comment (gary-madden) [inline .github/ISSUE_TEMPLATE/bug_report.md:25]: Copy pasta

File: .github/ISSUE_TEMPLATE/documentation-issue.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+about: Missing, misleading or incomplete documentation issues
+title: ''
+labels: docs
+assignees: ''
+
+---
+
+**Documentation issue**
+<!-- Before opening a new issue, please make sure you already checked -->
+<!-- our exhaustive wiki documentation at https://github.com/nosto/nosto-magento2-cmp/wiki -->
```
Comment (gary-madden) [inline .github/ISSUE_TEMPLATE/documentation-issue.md:12]: Copy pasta

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+## How Has This Been Tested?
+<!--- Please describe in detail how you tested your changes. -->
+<!--- Include details of your testing environment, and the tests you ran to -->
+<!--- see how your change affects other areas of the code, etc. -->
+
+## Documentation:
+<!--- Upon PR's approval, link the wiki page for your corresponding changes here. -->
+
+## Checklist:
+- [ ] My code follows the code style of this project.
```
Comment (gary-madden) [inline .github/pull_request_template.md:24]: We have a GitHub action to check this. Is it necessary to manually have a checkbox?

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] I have updated the documentation accordingly.
+- [ ] All new and existing tests passed.
```
Comment (gary-madden) [inline .github/pull_request_template.md:26]: We have a GitHub action for this. Is it necessary to manually have a checkbox?

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] I have assigned the correct milestone or created one if non-existent.
+- [ ] I have correctly labeled this pull request.
+- [ ] I have linked the corresponding issue in this description.
+- [ ] I have updated the corresponding Jira ticket.
```
Comment (gary-madden) [inline .github/pull_request_template.md:30]: There's no Jira for this.

File: build.gradle [hunk 8]
```diff
@@ -1,38 +1,61 @@
...
-    ignoreFailures = project.hasProperty("ignoreCheckFailures")
-}
-
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    withJavadocJar()
+    withSourcesJar()
+    sourceCompatibility = JavaVersion.VERSION_1_9
+    targetCompatibility = JavaVersion.VERSION_1_9
```
Comment (gary-madden) [inline build.gradle:38]: This would make it incompatible with one of our internal projects.

File: build.gradle [hunk 9]
```diff
@@ -1,38 +1,61 @@
...
+    mavenCentral()
+    jcenter()
+    maven {
+        url 'https://dl.bintray.com/palantir/releases'
+    }
 }
 
 dependencies {
-    testCompile group: 'junit', name: 'junit', version: '4.12'
+    testImplementation 'junit:junit:4.12'
```
Comment (gary-madden) [inline build.gradle:58]: This is a mixture of styles. I'd rather keep this consistent.

File: build.gradle [hunk 10]
```diff
@@ -45,29 +68,131 @@ dependencies {
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.9.9'
     compile group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: '2.9.9'
     compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.9'
+    errorprone "com.uber.nullaway:nullaway:0.7.9"
+    errorprone "com.google.errorprone:error_prone_core:2.3.4"
+    spotbugsPlugins 'com.h3xstream.findsecbugs:findsecbugs-plugin:1.10.1'
```
Comment (gary-madden) [inline build.gradle:73]: Same here. Inconsistent style.

File: build.gradle [hunk 11]
```diff
@@ -45,29 +68,131 @@ dependencies {
...
-repositories {
-    mavenCentral()
-    jcenter()
-    maven {
-        url 'https://dl.bintray.com/palantir/releases'
+// Configuration for the nullaway extension. The rest of the parameters must be
+// specified on the java-compile tasks. The only option that be specified here
+// is the name of the root package to be analysed.
+nullaway {
+    annotatedPackages.add("com.nostod")
```
Comment (gary-madden) [inline build.gradle:80]:
```suggestion
    annotatedPackages.add("com.nosto")
```

Looks like a typo here.

File: .github/pull_request_template.md (hunk 5 above)
Comment (mridang) [reply .github/pull_request_template.md:24]: Fixed.

File: .github/pull_request_template.md (hunk 6 above)
Comment (mridang) [reply .github/pull_request_template.md]: Fixed.

File: .github/pull_request_template.md (hunk 7 above)
Comment (mridang) [reply .github/pull_request_template.md]: Fixed.

File: build.gradle (hunk 9 above)
Comment (mridang) [reply build.gradle]: Fixed.

File: build.gradle (hunk 10 above)
Comment (mridang) [reply build.gradle]: Fixed.

File: build.gradle (hunk 8 above)
Comment (mridang) [reply build.gradle]: Fixed.

File: .github/ISSUE_TEMPLATE/documentation-issue.md (hunk 4 above)
Comment (mridang) [reply .github/ISSUE_TEMPLATE/documentation-issue.md]: Fixed.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (mridang) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Fixed.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (mridang) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Fixed.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (mridang) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Fixed.

File: build.gradle (hunk 11 above)
Comment (mridang) [reply build.gradle]: Fixed.

File: .github/pull_request_template.md (hunk 5 above)
Comment (gary-madden) [reply .github/pull_request_template.md:24]: Doesn't look fixed.

**Review by gary-madden — APPROVED** [review]
Just 1 minor quibble about the checkstyle checkbox but apart from that looks good 👍

File: .github/pull_request_template.md (hunk 5 above)
Comment (mridang) [reply .github/pull_request_template.md]: Fixed. My bad.

**Pull request merged**